### PR TITLE
Add support to checkin and view CDL items

### PR DESCRIPTION
**Pull request by mejackreed**

### CODE REVIEW

File: app/views/requests/index.html.erb [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <%= render partial: 'shared/group_tabs', locals: { self_value: patron.requests.length, group_value: patron.group.requests.length } if patron_or_group.group? %>
 <% end %>
 
-<% ready, requested = @requests.partition(&:ready_for_pickup?) %>
+<% ready, requested = @requests.select.partition { |request| request.ready_for_pickup? } %>
```
Comment (cbeer) [inline app/views/requests/index.html.erb:8]:
```suggestion
<% ready, requested = @requests.partition(&:ready_for_pickup?) %>
```

File: app/views/summaries/_summary.html.erb [hunk 2]
```diff
@@ -1,5 +1,6 @@
+<% active_cdl_checkouts, _inactive_cdl_checkouts = patron.requests.partition { |request| request.cdl_checkedout? } %>
```
Comment (cbeer) [inline app/views/summaries/_summary.html.erb:1]:
```suggestion
<% active_cdl_checkouts = patron.requests.select(&:cdl_checkedout?) %>
```

File: app/services/symphony_client.rb [hunk 3]
```diff
@@ -77,6 +77,24 @@ def patron_info(patron_key, item_details: {})
   end
   # rubocop:enable Metrics/MethodLength
 
+  # rubocop:disable Metrics/MethodLength
+  def circ_record_info(circ_record_key)
+    response = authenticated_request("/circulation/circRecord/key/#{circ_record_key}", params: {
+      includeFields: [
+        '*',
+        'circulationRule{loanPeriod{periodType{key}},renewFromPeriod}',
```
Comment (cbeer) [inline app/services/symphony_client.rb:85]: Is this still used?

File: app/models/request.rb [hunk 4]
```diff
@@ -110,6 +115,63 @@ def date_sort_key
...
+
+    @circ_record ||= begin
+      record = Checkout.find(cdl_circ_record_key, cdl: true)
+      return unless record.checkout_date == cdl_circ_record_checkout_date
+
+      record
+    end
+  end
+
+  def cdl?
```
Comment (cbeer) [inline app/models/request.rb:129]: Can we order these based on the offset in the CDL comment?

File: spec/models/request_spec.rb [hunk 5]
```diff
@@ -141,4 +141,28 @@
       expect(request).to be_from_borrow_direct
     end
   end
+
+  context 'when item is a CDL item' do
```
Comment (cbeer) [inline spec/models/request_spec.rb:145]: Should we import more comment tests?

File: app/models/request.rb (hunk 4 above)
Comment (mejackreed) [reply app/models/request.rb:129]: Reordered.

File: app/services/symphony_client.rb (hunk 3 above)
Comment (mejackreed) [reply app/services/symphony_client.rb]: Not that I can find, removed

File: spec/models/request_spec.rb (hunk 5 above)
Comment (mejackreed) [reply spec/models/request_spec.rb:145]: added

**Pull request merged**